components/micro-gateway-core/src/main/ballerina/src/gateway/filters/pre_authn_filter.bal

```diff
@@ -98,6 +100,33 @@ returns boolean {
                 authHeader = authCookie;
             }
         }
+    } else {
+        //process apikey authentication
+        if (authProvidersIds.indexOf(AUTH_SCHEME_API_KEY) != ()) {
+            json[] apiKeys = getAPIKeysforResource(context.getServiceName(), context.getResourceName());
+            printDebug(KEY_PRE_AUTHN_FILTER, apiKeys.toString());
+            if (apiKeys.length() > 0) {
+                foreach json apiKey in apiKeys {
+                    map<json> apiKeyMap = <map<json>>apiKey;
```

Comment:
Lets not cast this. Lets try to use
if(apiKey is map<json> ) {
}

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/932/commits/80d66417f6156d77e0396413735e126f5b839df6